### PR TITLE
[TK-06838] Close websocket sender

### PR DESCRIPTION
**Pull request by freesig**
This attempts to solve two issues with websockets.
1. When too many sockets are opened on the same port without closing (this is limited by ulimit) but we still don't want it unbounded. So it sends but connections full if there's more then 400. We can choose a different limit but 400 connections to one admin port seems like a lot and the default ulimit is 1024 but websockets require 2 file handles.
2. When connections are closed incorrectly by external programs the receiver is closed but not the sender leading to file handles collecting over time. This solves that by closing the sender as the receiver is closed. @zippy this is what you were running into.

### CODE REVIEW

File: crates/holochain/src/conductor/interface/websocket.rs [hunk 1]
```diff
@@ -171,11 +183,28 @@ async fn handle_shutdown(listener_handles: Vec<JoinHandle<InterfaceResult<()>>>)
...
-            Err(InterfaceError::Closed) => break,
-            Err(e) => error!(error = &e as &dyn std::error::Error),
+            Err(InterfaceError::Closed) => {
+                if let Err(e) =
+                    WebsocketSender::close(&mut tx_to_iface, 1000, "Shutting down".into()).await
+                {
+                    warn!("Admin socket failed to close: {}", e);
+                }
+                if num_connections.load(Ordering::SeqCst) > 0 {
+                    num_connections.fetch_sub(1, Ordering::SeqCst);
```
Comment (neonphog) [inline crates/holochain/src/conductor/interface/websocket.rs:201]: Alas, this is not atomic... it could change between these operations

Comment (maackle) [reply crates/holochain/src/conductor/interface/websocket.rs:201]: Is it even possible for the connection count to go negative?

Even if so, this non-atomic check can't hurt, it seems like a sanity check that might fail in an extreme race condition. But I wonder if it's necessary at all.

File: Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [workspace]
 members = [
+  # "crates/admin",
```
Comment (maackle) [inline Cargo.toml:3]: What is this admin crate that keeps popping up as a comment?

**Review by maackle — APPROVED** [review]

File: crates/holochain/src/conductor/interface/websocket.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain/src/conductor/interface/websocket.rs:201]: yeh how are you meant to do checked subs with atomics?

File: Cargo.toml (hunk 2 above)
Comment (freesig) [reply Cargo.toml:3]: It's the holochain_admin crate I've made to launch / configure conductors. I'd like to pull it out of the workspace but need the interface types to do so. So probably can now.

File: crates/holochain/src/conductor/interface/websocket.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain/src/conductor/interface/websocket.rs]: I think this is better now

**Review by neonphog — APPROVED** [review]

**Pull request merged**